docs/the-new-architecture/pillars-fabric-components.md

```diff
@@ -296,8 +296,237 @@ The `gradle` contains various section to properly configure the library and to a
+            │   └── FBReactNativeSpec.h
+            ├── RCTThirdPartyFabricComponentsProvider.h
+            ├── RCTThirdPartyFabricComponentsProvider.mm
+            └── react
+                └── renderer
+                    └── components
+                        ├── RNTCenteredTextSpecs
+                        │   ├── ComponentDescriptors.h
+                        │   ├── EventEmitters.cpp
+                        │   ├── EventEmitters.h
+                        │   ├── Props.cpp
+                        │   ├── Props.h
+                        │   ├── RCTComponentViewHelpers.h
+                        │   ├── ShadowNodes.cpp
+                        │   └── ShadowNodes.h
+                        └── rncore
+                            ├── ComponentDescriptors.h
+                            ├── EventEmitters.cpp
+                            ├── EventEmitters.h
+                            ├── Props.cpp
+                            ├── Props.h
+                            ├── RCTComponentViewHelpers.h
+                            ├── ShadowNodes.cpp
+                            └── ShadowNodes.h
+```
+
+The relevant path for the Component we are writing is `generated/build/generated/ios/react/renderer/components/RNTCenteredTextSpecs`.
+This folder contains all the genereted code required by our Component.
+
+See the [CodeGen](./pillars-codegen) section for further details on the generated files.
+
+##### Write the Native iOS Code
+
+Now that we can see the iOS code we need, it's time to write the Native code for our Fabric Component.
+We need to create three files:
+
+1. The `RNTCenteredTextManager.mm`, an Objective-C++ file which declares what the Component exports.
+2. The `RNTCenteredText.h`, an header file for the actual view.
+3. The `RNTCenteredText.mm`, the implementation of the view.
+
+###### RNTCenteredTextManager.mm
+
+```obj-c title="RNTCenteredTextManager.mm"
+#import <React/RCTLog.h>
+#import <React/RCTUIManager.h>
+#import <React/RCTViewManager.h>
+
+@interface RTNCenteredTextManager : RCTViewManager
+@end
+
+@implementation RTNCenteredTextManager
+
+RCT_EXPORT_MODULE(RTNCenteredText)
+
+RCT_EXPORT_VIEW_PROPERTY(text, NSString)
+
+- (UIView *)view
+{
+  return [[UIView alloc] init];
+}
+
+@end
+```
+
+This file is the manager for the Fabric Component.
+
+The most important call is to the `RCT_EXPORT_MODULE` which is required to export the module so that fabric can actually retrieve it and instantiate it.
+
+Then, we have to expose a property for the Fabric Component. This is done with the `RCT_EXPORT_VIEW_PROPERTY` macro, specifieng a name and a type.
+
+Finally, we need to add a `- ((UIView *)view)` method for legacy reasons.
+
+:::info
+There are other macros that can be used to export custom properties, emitters and other constructs. You can look them up [here](https://github.com/facebook/react-native/blob/main/React/Views/RCTViewManager.h)
+:::
+
+###### RNTCenteredText.h
+
+```Objective-C title="RNTCenteredText.h"
```

Comment:
```suggestion
```Objective-C title="RNTCenteredText.h"
```

Since the preview deploy failed, I'm not able to verify this, but languge name for the code block might be incorrect.

Generally, it's safe to use language keywords listed on the Prism site - https://prismjs.com/#supported-languages - however, React Prism which is used underneath in Dousauaurs implementation might not support 100% of the list so it's always nice to verify the language name, or just use the already present on the website language keyword.

Reply:
thanks for the feedback!